### PR TITLE
CLOUDP-237487 Add metadata file to control versions

### PR DESCRIPTION
**Pull request by fmenezes**
- Added a metadata.json file to store versions
- Added another make task `build-all` to build against each MongoDB version

### CODE REVIEW

File: scripts/build-all.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+for VERSION in $(jq -r '.mongodb_versions | .[]' < ./scripts/manifest.json); do
+  ./scripts/build.sh  --mongodb-version "$VERSION" \
+                      --mongosh-version "$MONGOSH_VERSION" \
+                      --tag "$TAG:$VERSION" \
+                      --platform "$PLATFORM"
+  MINOR_VERSION=$(echo "$VERSION" | awk 'BEGIN {FS="."}{print $1 "." $2}')
+  MAJOR_VERSION=$(echo "$VERSION" | awk 'BEGIN {FS="."}{print $1}')
+  docker tag "$TAG:$VERSION" "$TAG:$VERSION-$DATE_TAG"
+  docker tag "$TAG:$VERSION" "$TAG:$MINOR_VERSION"
+  docker tag "$TAG:$VERSION" "$TAG:$MAJOR_VERSION"
```
Comment (jeroenvervaeke) [inline scripts/build-all.sh:35]: This script only works when the `mongodb_versions` in `scripts/manifest.json` are sorted.

Should we make this more robust/add a check (github action) that all versions are sorted correctly?

File: scripts/manifest.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (jeroenvervaeke) [inline scripts/manifest.json:1]: Would it make more sense to have the manifest in the root folder of the repository?

File: Makefile [hunk 3]
```diff
@@ -1,6 +1,7 @@
 MONGODB_VERSION ?= 7.0.6
```
Comment (jeroenvervaeke) [inline Makefile:1]: Instead of having a hardcoded version here, can we use the manifest file too?

File: scripts/build-all.sh (hunk 1 above)
Comment (fmenezes) [reply scripts/build-all.sh:35]: I've added sort on jq

Comment (jeroenvervaeke) [reply scripts/build-all.sh:37]: This won't work, as it'll sort the versions lexicographically.

Example:
```json
    "mongodb_versions": [
      "7.0.0",
      "7.0.1",
      "7.0.2",
...
      "7.0.8",
      "7.0.9",
      "7.0.10",
      "7.0.11"
    ]
```

Result:
```plain
7.0.0
7.0.1
7.0.10
7.0.11
7.0.2
...
```

This should do the trick:
```sh
jq -r '.mongodb_versions | sort_by( split(".") | map(tonumber) ) | .[]' < ./manifest.json
```

File: scripts/build.sh [hunk 4]
```diff
@@ -30,18 +32,15 @@ while [[ $# -gt 0 ]]; do
 done
 
 if [[ -z "$MONGODB_VERSION" ]]; then
-  echo "--mongodb-version is missing"
-  exit 1
+  MONGODB_VERSION=$(jq -r '.mongodb_versions | sort | last' < "$MANIFEST_FILE")
```
Comment (jeroenvervaeke) [inline scripts/build.sh:35]: We'll also have to sort here using ` | sort_by( split(".") | map(tonumber) ) | `

**Review by jeroenvervaeke — CHANGES_REQUESTED** [review]

File: scripts/build-all.sh (hunk 1 above)
Comment (fmenezes) [reply scripts/build-all.sh:37]: good catch

**Review by jeroenvervaeke — APPROVED** [review]

**Pull request merged**